src/openzeppelin/tests/access/test_dual_accesscontrol.cairo

```diff
@@ -0,0 +1,314 @@
+use traits::Into;
+use array::ArrayTrait;
+use starknet::ContractAddress;
+use starknet::contract_address_const;
+use starknet::testing::set_caller_address;
+use starknet::testing::set_contract_address;
+use zeroable::Zeroable;
+
+// Dispatchers
+use openzeppelin::access::accesscontrol::interface::IAccessControlDispatcher;
+use openzeppelin::access::accesscontrol::interface::IAccessControlCamelDispatcher;
+use openzeppelin::access::accesscontrol::interface::IAccessControlDispatcherTrait;
+use openzeppelin::access::accesscontrol::interface::IAccessControlCamelDispatcherTrait;
+
+// Dual case AccessControl
+use openzeppelin::access::accesscontrol::dual_accesscontrol::DualCaseAccessControlTrait;
+use openzeppelin::access::accesscontrol::dual_accesscontrol::DualCaseAccessControl;
+
+// Mocks
+use openzeppelin::tests::mocks::snake_accesscontrol_mock::SnakeAccessControlMock;
+use openzeppelin::tests::mocks::camel_accesscontrol_mock::CamelAccessControlMock;
+use openzeppelin::tests::mocks::accesscontrol_panic_mock::SnakeAccessControlPanicMock;
+use openzeppelin::tests::mocks::accesscontrol_panic_mock::CamelAccessControlPanicMock;
+use openzeppelin::tests::mocks::non_implementing_mock::NonImplementingMock;
+
+// Other
+use openzeppelin::tests::utils;
+use openzeppelin::tests::utils::PanicTrait;
+use openzeppelin::access::accesscontrol::DEFAULT_ADMIN_ROLE;
+
+///
+/// Constants
+///
+
+fn ADMIN() -> ContractAddress {
+    contract_address_const::<10>()
+}
+
+fn AUTHORIZED() -> ContractAddress {
+    contract_address_const::<20>()
+}
+
+const ROLE: felt252 = 41;
+
+///
+/// Setup
+///
+
+fn setup_snake() -> (DualCaseAccessControl, IAccessControlDispatcher) {
+    let mut calldata = ArrayTrait::new();
+    calldata.append(ADMIN().into());
+    //set_caller_address(OWNER());
```

Comment:
This is commented because later you can't use `set_contract_address(OWNER())` before protected calls because of the potential compiler bug, right?

Reply:
Ah whoops. I was contemplating setting the admin with `get_caller_address` and decided against it. I forgot to remove these 😅